### PR TITLE
slade: 3.1.1.5 -> 3.2.1

### LINKED ISSUE #193079: `slade` is out-of-date
###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: slade

current version: 3.1.1.5
desired version: 3.2.1

###### Notify maintainers

maintainers: @abbradar

###### Note for maintainers

Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by iAmMrinal0**
###### Description of changes

Package update: [Changelog: v3.2.1](https://github.com/sirjuddington/SLADE/releases/tag/3.2.1)
Closes #193079 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/slade/default.nix [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
+    sha256 = "sha256-KFRX3sfI//Op/h/EfEuAZOY22RO5qNXmvhSksC0aS4U=";
   };
 
   nativeBuildInputs = [ cmake pkg-config zip ];
-  buildInputs = [ wxGTK gtk2 sfml fluidsynth curl freeimage ftgl glew ];
+  buildInputs =
+    [ wxGTK gtk2 sfml fluidsynth curl freeimage ftgl glew lua mpg123 ];
+
+  cmakeFlags = [
+    "-DwxWidgets_CONFIG_EXECUTABLE=${wxGTK}/bin/wx-config"
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/slade/default.nix:20]: This is not cross friendly. Can we use it from PATH instead?

File: pkgs/applications/misc/slade/default.nix [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
     repo = "SLADE";
     rev = version;
-    sha256 = "0mdn59jm6ab4cdh99bgvadif3wdlqmk5mq635gg7krq35njgw6f6";
+    sha256 = "sha256-KFRX3sfI//Op/h/EfEuAZOY22RO5qNXmvhSksC0aS4U=";
   };
 
   nativeBuildInputs = [ cmake pkg-config zip ];
-  buildInputs = [ wxGTK gtk2 sfml fluidsynth curl freeimage ftgl glew ];
+  buildInputs =
+    [ wxGTK gtk2 sfml fluidsynth curl freeimage ftgl glew lua mpg123 ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/slade/default.nix:17]:
```suggestion
  buildInputs = [
    wxGTK
    gtk2
    sfml
    fluidsynth
    curl
    freeimage
    ftgl
    glew
    lua
    mpg123
  ];
```

File: pkgs/applications/misc/slade/default.nix [hunk 3]
```diff
@@ -1,18 +1,26 @@
...
+
+  cmakeFlags = [
+    "-DwxWidgets_CONFIG_EXECUTABLE=${wxGTK}/bin/wx-config"
+    "-DWX_GTK3=OFF"
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/slade/default.nix:21]: Why not use gtk3? Gtk2 is deprecated.

File: pkgs/applications/misc/slade/default.nix (hunk 1 above)
Comment (iAmMrinal0) [reply pkgs/applications/misc/slade/default.nix:20]: do you have any pointers on how to do that?

Comment (SuperSandro2000) [reply pkgs/applications/misc/slade/default.nix:20]: I would need to go into the source code or we use buildPackages.

Comment (wegank) [reply pkgs/applications/misc/slade/default.nix:20]: In fact, we only have to add `which` in `nativeBuildInputs`, see [here](https://github.com/sirjuddington/SLADE/blob/master/src/CMakeLists.txt#L20).

**Comment by wegank** [comment]
Closed by #198185.

**Pull request closed without merging**